### ISSUE TITLE
After Rebase, warn against Sync or Pull and recommend Push (Force)

### ISSUE THREAD

**Issue by AnthonyMastrean** [microsoft/vscode#193143]
After rebasing a branch onto its parent, Git `status` reports that the local and remote branches have "diverged" and recommends pulling. This is inappropriate advice at this point in the "rebase onto" workflow. If followed, it can be quite a headache to repair. I still look up the `reflog` commands every time.

```
$ git status
On branch rebase-example
Your branch and 'origin/rebase-example' have diverged,
and have 2 and 9 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)

nothing to commit, working tree clean
```

Visual Studio Code follows suit, shows diverged histories, and recommends syncing. This is worse than Git's behavior because this will mangle the remote branch, too!

<img width="512" alt="image" src="https://github.com/microsoft/vscode/assets/217842/14cad4f9-3595-4cdc-beab-eeec7f52d928">

There's a warning if you click Sync, but I dismissed this years ago during a routine Sync.

<img width="512" alt="image" src="https://github.com/microsoft/vscode/assets/217842/5f6214f8-cafd-4dd1-bbe9-bc45b9f964c7">

There's a Git (Extension) pop-up that detects attempts to pull after "rebase onto", but it doesn't say why pull might be a bad idea nor does it offer the correct action Push (Force). I'd argue that highlighting "Always Pull" as the preferred choice is also a bug.

<img width="512" alt="image" src="https://github.com/microsoft/vscode/assets/217842/687ab2d0-ffb4-443d-aea1-1871bb7769ed">

Even if I click "Don't Pull", an error dialog encouraging me to pull is shown!

<img width="512" alt="image" src="https://github.com/microsoft/vscode/assets/217842/ee097019-369a-46e3-9717-7120513690a9">

The only correct action after an intentional and successful "rebase onto" is a Push (Force). It would be awesome to see that action in the blue button on the Source Control panel.

**Comment by Malix-off** [comment]
Fell for this trap a pretty huge amount of time, thinking it would obviously `push --force-with-lease` by default for the next push after the rebase, and thinking my coworkers had pushed some commits in the meantime

**Comment by luckielordie** [comment]
Yeah, I fell for this one after a Rebase too. It would be nice if the UI provided some feedback on what I was meant to do next.

**Comment by ecl1ps** [comment]
This would be great. Finding the hidden force push menu item after each rebase is a PITA.
